### PR TITLE
Decrease memory usage when load disk-ann index (#20886)

### PR DESCRIPTION
**Pull request by zhagnlu**
#20886 
Signed-off-by: zhagnlu <lu.zhang@zilliz.com>

### CODE REVIEW

**Comment by xige-16** [comment]
tips: DiskFileManager.AddFile also has the risk of memory overflow

**Comment by zhagnlu** [comment]
rerun ut

**Comment by zhagnlu** [comment]
rerun ut

File: configs/milvus.yaml [hunk 1]
```diff
@@ -393,6 +393,8 @@ common:
     BeamWidthRatio: 4.0
   # This parameter specify how many times the number of threads is the number of cores
   threadCoreCoefficient : 10
+  # This parameter specify how much memory when caching index to local disk
+  diskIndexMaxMemoryLimit : 20971520 # KB
```
Comment (czs007) [inline configs/milvus.yaml:397]: better to be MB

**Comment by zhagnlu** [comment]
run-cpu-e2e

**Pull request closed without merging**